### PR TITLE
Add a simple build and test CI

### PR DESCRIPTION
**Pull request by j2kun**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Build and Test
+on: [push]
```
Comment (asraa) [inline .github/workflows/build_and_test.yml:2]: nit: recommend adding default read-only perms at the top-level of the workflow

```
permissions: read-all
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "This job was triggered by a ${{ github.event_name }} event."
+      - run: echo "Running on OS ${{ runner.os }}"
+      - run: echo "Checking out branch ${{ github.ref }}."
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: "Run `bazel build`"
+        run: |
+          bazel build //...
```
Comment (asraa) [inline .github/workflows/build_and_test.yml:14]: Is bazel already installed? I don't *think* it is, and you might need a bazel installation action

https://github.com/bazelbuild/setup-bazelisk maybe this?

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+on: [push]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "This job was triggered by a ${{ github.event_name }} event."
+      - run: echo "Running on OS ${{ runner.os }}"
+      - run: echo "Checking out branch ${{ github.ref }}."
+      - name: Check out repository code
+        uses: actions/checkout@v3
```
Comment (asraa) [inline .github/workflows/build_and_test.yml:11]: nit: in a follow-up I can pin actions by hashes & use depandabot to update them

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (j2kun) [reply .github/workflows/build_and_test.yml:14]: It is installed! https://github.com/actions/runner-images/pull/490

**Comment by j2kun** [comment]
The action succeeded https://github.com/google/heir/actions/runs/5126658016/jobs/9221436296

Though it took 1.5 hours. @asraa what do you think about merging this as-is and, by possibly ignoring it for internal CLs, keep going and have a followup ticket to use a hosted runner and/or cache. I want to do a bit more research into how the caching works in bazel and what other MLIR Google projects do.

**Comment by asraa** [comment]
> keep going and have a followup ticket to use a hosted runner and/or cache.

Yes! Check out the GitHub Action cache - if that's not good enough, then probably RBE is the way.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (asraa) [reply .github/workflows/build_and_test.yml:14]: thanks!

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (j2kun) [reply .github/workflows/build_and_test.yml]: just did it, using pin-github-action https://github.com/mheap/pin-github-action, manually confirming it's the right hash https://github.com/actions/checkout/commit/8e5e7e5ab8b370d6c329ec480221332ada57f0ab

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (j2kun) [reply .github/workflows/build_and_test.yml]: Done

**Pull request merged**